File: chessboard.java
int ascii_lowercase = "abcdefghijklmnopqrstuvwxyz"
class chessboard()
{
  private Object[][] populate(Object[][] arr)
  {
    arr[0][0] = new rook("white", [0,0]);
    arr[0][1] = new knight("white", [0,1]);
    arr[0][2] = new bishop("white", [0,2]);
    arr[0][3] = new queen("white", [0,3]);
    arr[0][4] = new king("white", [0,4]);
    arr[0][5] = new bishop("white", [0,5]);
    arr[0][6] = new knight("white", [0,6]);
    arr[0][7] = new rook("white", [0,7]);
    for (int i = 0; i < 8; i++)
    {
      arr[1][i] = new pawn("white", [1,i]);
    }
    for (int i = 0; i < 8; i++)
    {
      arr[6][i] = new pawn("black", [6,i]);
    }
    arr[7][0] = new rook("black", [7,0]);
    arr[7][1] = new knight("black", [7,1]);
    arr[7][2] = new bishop("black", [7,2]);
    arr[7][3] = new queen("black", [7,3]);
    arr[7][4] = new king("black", [7,4]);
    arr[7][5] = new bishop("black", [7,5]);
    arr[7][6] = new knight("black", [7,6]);
    arr[7][7] = new rook("black", [7,7]);
    return arr;
  }
  
  public chessboard()
  {
    private Object[][] this.chessboard = new Object[8][8];
    this.chessboard.populate();
  }

  public move()
  {
    piece.move
  }
}

File: VScode/Personal Projects/Chess Website/chessboard.js
import {bishop} from './chess_pieces_js/bishop.js';
import {empty} from './chess_pieces_js/empty.js';
import {king} from './chess_pieces_js/king.js';
import {knight} from './chess_pieces_js/knight.js';
import {pawn} from './chess_pieces_js/pawn.js';
import {queen} from './chess_pieces_js/queen.js';
import {rook} from './chess_pieces_js/rook.js';

export class chessboard {
  constructor () {
    var a1 = new rook();
    var b1 = new knight();
    var c1 = new bishop();
    var d1 = new queen();
    var e1 = new king();
    var f1 = new bishop();
    var g1 = new knight();
    var h1 = new rook();

    var a2 = new pawn();
    var b2 = new pawn();
    var c2 = new pawn();
    var d2 = new pawn();
    var e2 = new pawn();
    var f2 = new pawn();
    var g2 = new pawn();
    var h2 = new pawn();

    var a3 = new empty();
    var b3 = new empty();
    var c3 = new empty();
    var d3 = new empty();
    var e3 = new empty();
    var f3 = new empty();
    var g3 = new empty();
    var h3 = new empty();
  
    var a4 = null;
    var b4 = null;
    var c4 = null;
    var d4 = null;
    var e4 = null;
    var f4 = null;
    var g4 = null;
    var h4 = null;
  
    var a5 = null;
    var b5 = null;
    var c5 = null;
    var d5 = null;
    var e5 = null;
    var f5 = null;
    var g5 = null;
    var h5 = null;
  
    var a6 = null;
    var b6 = null;
    var c6 = null;
    var d6 = null;
    var e6 = null;
    var f6 = null;
    var g6 = null;
    var h6 = null;
  
    var a7 = new pawn();
    var b7 = new pawn();
    var c7 = new pawn();
    var d7 = new pawn();
    var e7 = new pawn();
    var f7 = new pawn();
    var g7 = new pawn();
    var h7 = new pawn();
  
    var a8 = new rook();
    var b8 = new knight();
    var c8 = new bishop();
    var d8 = new queen();
    var e8 = new king();
    var f8 = new bishop();
    var g8 = new knight();
    var h8 = new rook();

    var chessboard = [
      [a1,b1,c1,d1,e1,f1,g1,h1],
      [a2,b2,c2,d2,e2,f2,g2,h2],
      [a3,b3,c3,d3,e3,f3,g3,h3],
      [a4,b4,c4,d4,e4,f4,g4,h4],
      [a5,b5,c5,d5,e5,f5,g5,h5],
      [a6,b6,c6,d6,e6,f6,g6,h6],
      [a7,b7,c7,d7,e7,f7,g7,h7],
      [a8,b8,c8,d8,e8,f8,g8,h8],
    ];
  }

  get_board() {
    return this.chessboard;
  }

  get_piece(x, y) {
    return chessboard[y][x];
  }
  move_logic(object_class, move) {
    var pos1 = move[0];
    var pos2 = move[1];
  }
}